### ISSUE TITLE
DeviceId `0` private

### ISSUE THREAD

**Issue by j4qfrost** [rust-windowing/winit#1413]
Is there a reason why the id number is left private? It should be safe the expose this right?

https://github.com/rust-windowing/winit/blob/3e3bb8a8f1b808424f2eaf262130f8830dedbaa2/src/platform_impl/linux/x11/mod.rs#L499

Alternatively, should a getter be implemented?

**Comment by j4qfrost** [comment]
I made a pull request on a remote branch here https://github.com/dvc94ch/winit/pull/1

**Comment by dvc94ch** [comment]
yes it would be useful to have `DeviceId` expose a `fn id(&self) -> u32`.

**Comment by j4qfrost** [comment]
https://github.com/rust-windowing/winit/pull/1421 New pull request @goddessfreya @dvc94ch

**Comment by dvc94ch** [comment]
I changed my mind. I'm not sure it's such a great idea anymore. Did you look at how I implemented the pointer handling without changing winit? There's no way around assigning our own ids to pointers, since flutter considers each finger to be a pointer even if it shares the device id. Also on Wayland the device id would correspond to a seat, which means that a pointer and a touchscreen share the same id.

**Comment by kchibisov** [comment]
>Also on Wayland the device id would correspond to a seat, which means that a pointer and a touchscreen share the same id.

You can actually use wl_pointer and wl_touch objects to distinguish them, there's an ongoing work about event redesign here anyway https://github.com/rust-windowing/winit/pull/1374
